packages/assets-controllers/src/Standards/NftStandards/ERC1155/ERC1155Standard.ts

```diff
@@ -161,13 +209,31 @@ export class ERC1155Standard {
     standard: string;
     tokenURI: string | undefined;
     image: string | undefined;
+    name: string | undefined;
+    symbol: string | undefined;
   }> => {
     const isERC1155 = await this.contractSupportsBase1155Interface(address);
 
     if (!isERC1155) {
       throw new Error("This isn't a valid ERC1155 contract");
     }
-    let tokenURI, image;
+
+    let tokenURI, image, symbol, name;
+
+    const results = await Promise.allSettled([
+      this.getAssetName(address),
+      this.getAssetSymbol(address),
+    ]);
+
+    const fulfilledResults = results.filter(
+      (result): result is PromiseFulfilledResult<string> =>
+        result.status === 'fulfilled',
+    );
```

Comment:
Since the [docs](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/allSettled#return_value) say `value: Only present if status is "fulfilled"`.  I'm wondering if the filtering is not required and you can just do:

```
{
  ...
  name: results[0].value,
  symbol: results[1].value,
}
```

Since it will already be undefined if it failed.  Would fix the other comments too.

Reply:
yes but types lint will fail ,because Promise.allSettled returns a union of either settled or rejected. The rejected one has no value property ( we have the reason property on that one ). since you did not narrow the union to the settled type, so the compiler correctly informs you about the invalid access.